Index: texlive-extra/debian/changelog
===================================================================
--- texlive-extra/debian/changelog	(revision 3046)
+++ texlive-extra/debian/changelog	(working copy)
@@ -1,4 +1,4 @@
-texlive-extra (2007-4~1) unstable; urgency=low
+texlive-extra (2007.1-1~1) unstable; urgency=low
 
   * add stricter dependencies to all packages: if in the final shipout
     dep on a texlive package there is no version given, the value of
@@ -6,9 +6,18 @@
   * move bigfoot from tl-humanities to tl-latex-extra (upstream move, too)
     Add a tl-latex-extra replaces tl-humanities and a tl-latex-extra
     recommends tl-humanities (Closes: #420394)
+  * Take over jadetex, xmltex and passivetex, with maintainers' approval
+    unless they are MIA: The new texlive-xml package declares
+    Conflicts/Replaces/Provides on these three packages.
 
- -- Norbert Preining <preining@debian.org>  Tue, 10 Jul 2007 00:48:38 +0200
+    These formats and executables are already integrated upstream, but had
+    been left out of the generated orig.tar.gz, hence we need a new
+    upstream version. [fk]
+  * move xmlplay back to texlive-xml, and let this one replace 
+    texlive-latex-extra (<= 2007-3) [np]
 
+ -- Norbert Preining <preining@debian.org>  Mon, 20 Aug 2007 18:42:31 +0200
+
 texlive-extra (2007-3) unstable; urgency=low
 
   * Fix the friday 13 bugs: A bad coincidence combined a serious bug in
Index: all/debian/CHANGES.packaging
===================================================================
--- all/debian/CHANGES.packaging	(revision 3047)
+++ all/debian/CHANGES.packaging	(working copy)
@@ -4,6 +4,20 @@
 # $Id$
 #
 
+2007-08-20
+  (merge from htmlxml branch, all work done by fk)
+  * Take over jadetex, xmltex and passivetex, with maintainers' approval
+    unless they are MIA: The new texlive-xml package declares
+    Conflicts/Replaces/Provides on these three packages.
+
+    These formats and executables are already integrated upstream, but had
+    been left out of the generated orig.tar.gz, hence we need a new
+    upstream version.
+  * move xmlplay back to texlive-xml, and let this one replace 
+    texlive-latex-extra (<= 2007-3) [np]
+
 2007-08-16
   * fix FTBFS on hurd-i386, adding patch icu-powerpc-buildfix-patch from
     Samuel Thibault (Closes: #437949) [np]
Index: all/debian/tpm2deb.cfg
===================================================================
--- all/debian/tpm2deb.cfg	(revision 3046)
+++ all/debian/tpm2deb.cfg	(working copy)
@@ -39,7 +39,7 @@
 # updated latest-version for texlive-lang, but NOT UPLOADED, and then
 # you build texlive-base and upload it, IT WILL BREAK!
 latest-version;texlive-base;2007-11
-latest-version;texlive-extra;2007
+latest-version;texlive-extra;2007.1
 latest-version;texlive-bin;2007-13
 latest-version;texlive-doc;2007
 latest-version;texlive-lang;2007
@@ -49,9 +49,10 @@
 name;collection-basicbin;texlive-base-bin;texlive-bin
 name;collection-binextra;texlive-extra-utils;texlive-bin
 name;collection-fontbin;texlive-font-utils;texlive-bin
-# htmlxml will not be build, but is replaced by xmtex, jadetex, and passivetex
-# see below
-#name;collection-htmlxml;texlive-htmlxml;texlive-bin
+# collection-htmlxml is now built as tl-xml to replace xmltex,
+# jadetex, and passivetex (not tex4ht).  Missing Dep on kvoptions.sty
+name;collection-htmlxml;texlive-xml;texlive-extra
+depends;texlive-xml;texlive-latex-recommended
 name;collection-metapost;texlive-metapost;texlive-bin
 name;collection-omega;texlive-omega;texlive-bin
 name;collection-xetex;texlive-xetex;texlive-bin
@@ -470,9 +471,11 @@
 #move;mltex;texlive-formats-extra
 depends;texlive-formats-extra;texlive-latex-base
 ####conflicts;texlive-formats-extra;tetex-bin (<= 3.0-13)
-# xmlplay *was* in texlive-htmlxml, but this is not build anymore, so move
-# it to latex-extra
-move;xmlplay;texlive-latex-extra
+# xmlplay originally was in texlive-htmlxml, but since this wasn't build
+# for a long time, it was moved to texlive-latex-extra. Now we build
+# texlive-xml and move it back there, since it depended anyway on xmltex.
+#move;xmlplay;texlive-latex-extra
+replaces;texlive-xml;texlive-latex-extra (<= 2007-3)
 # move yi4latex, the only package missing in the latex-cjk stuff, to
 # texlive-latex-extra
 move;yi4latex;texlive-latex-extra
@@ -753,11 +756,19 @@
 # this is its own collection which is not included, see NAME section above
 recommends;texlive-full;latex-cjk-all (>= 4.6.0+cvs20060714-2)
 #
-# jadetex, xmltex, passivetex
-# all packages which were contained in texlive-xmlhtml
-# this collection is not included, see NAME section above
-suggests;texlive-full;jadetex (>= 3.13-7.1)
-suggests;texlive-full;xmltex (>> 1.9-11.1), passivetex (>> 1.25-2)
+# jadetex, xmltex, passivetex takeover
+replaces;texlive-xml;jadetex
+conflicts;texlive-xml;jadetex
+provides;texlive-xml;jadetex
+replaces;texlive-xml;xmltex
+conflicts;texlive-xml;xmltex
+provides;texlive-xml;xmltex
+replaces;texlive-xml;passivetex
+conflicts;texlive-xml;passivetex
+provides;texlive-xml;passivetex
+# do not take over tex4ht
+blacklist;tpm;tex4ht;*
+blacklist;tpm;bin-tex4ht;*
 #
 # dviutils
 # dviconcat and dviselect are in dvitutils
@@ -843,6 +854,11 @@
 blacklist;file;bin/[^/]*/texsis
 # tl-math-extra
 blacklist;file;bin/[^/]*/amstex
+# tl-xml
+blacklist;file;bin/[^/]*/jadetex
+blacklist;file;bin/[^/]*/pdfjadetex
+blacklist;file;bin/[^/]*/xmltex
+blacklist;file;bin/[^/]*/pdfxmltex
 # tl-lang-cz
 blacklist;file;bin/[^/]*/cslatex
 blacklist;file;bin/[^/]*/pdfcslatex
